beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -491,6 +535,107 @@ impl<T: EthSpec> ValidatorMonitor<T> {
+            .map(|(slot, attestation)| (*slot, (*attestation).clone()))
+            .collect();
+
+        let unaggregated_attestations = &mut self.unaggregated_attestations;
+        for (slot, unaggregated_attestation) in attested_slots {
+            if unaggregated_attestations.remove_entry(&slot).is_some() {
+                // Don't process this attestation, it's too old to be processed by this state.
+                if slot.epoch(T::slots_per_epoch()) < state.previous_epoch() {
+                    continue;
+                }
+
+                // We are simulating that unaggregated attestation in a service that produces unaggregated attestations
+                // every slot, the inclusion_delay shouldn't matter here as long as the minimum value
+                // that qualifies the committee index for reward is included
+                let inclusion_delay = spec.min_attestation_inclusion_delay;
+
+                // Get the reward indices for the unaggregated attestation or log an error
+                match get_attestation_participation_flag_indices(
+                    state,
+                    &unaggregated_attestation.data,
+                    inclusion_delay,
+                    spec,
+                ) {
+                    Ok(flag_indices) => {
+                        let head_hit = flag_indices.get(TIMELY_HEAD_FLAG_INDEX);
+                        let target_hit = flag_indices.get(TIMELY_TARGET_FLAG_INDEX);
+                        let source_hit = flag_indices.get(TIMELY_SOURCE_FLAG_INDEX);
+
+                        if head_hit.is_some() {
+                            metrics::inc_counter(
+                                &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_HEAD_ATTESTER_HIT,
+                            );
+                        } else {
+                            metrics::inc_counter(
+                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_HEAD_ATTESTER_MISS,
+                            );
+                        }
+                        if target_hit.is_some() {
+                            metrics::inc_counter(
+                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_TARGET_ATTESTER_HIT,
+                            );
+                        } else {
+                            metrics::inc_counter(
+                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_TARGET_ATTESTER_MISS,
+                            );
+                        }
+                        if source_hit.is_some() {
+                            metrics::inc_counter(
+                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_SOURCE_ATTESTER_HIT,
+                            );
+                        } else {
+                            metrics::inc_counter(
+                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_SOURCE_ATTESTER_MISS,
+                            );
+                        }
```

Comment:
```suggestion
                        let head_hit = flag_indices.contains(TIMELY_HEAD_FLAG_INDEX);
                        let target_hit = flag_indices.contains(TIMELY_TARGET_FLAG_INDEX);
                        let source_hit = flag_indices.contains(TIMELY_SOURCE_FLAG_INDEX);

                        if head_hit {
                            metrics::inc_counter(
                                &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_HEAD_ATTESTER_HIT,
                            );
                        } else {
                            metrics::inc_counter(
                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_HEAD_ATTESTER_MISS,
                            );
                        }
                        if target_hit {
                            metrics::inc_counter(
                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_TARGET_ATTESTER_HIT,
                            );
                        } else {
                            metrics::inc_counter(
                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_TARGET_ATTESTER_MISS,
                            );
                        }
                        if source_hit {
                            metrics::inc_counter(
                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_SOURCE_ATTESTER_HIT,
                            );
                        } else {
                            metrics::inc_counter(
                            &metrics::VALIDATOR_MONITOR_ATTESTATION_SIMULATOR_SOURCE_ATTESTER_MISS,
                            );
                        }
```

These flag indices are a bit tricky, but we should be checking to see if the flag indices *contain a value* rather than *have a value at an index.* Here's an example of some existing consensus code: https://github.com/v4lproik/lighthouse/blob/00b439ee21875400745c177274b228081efa8148/consensus/state_processing/src/per_block_processing/process_operations.rs#L160